sdk-ext-lib/src/main/java/cash/z/ecc/sdk/SynchronizerCompanion.kt

```diff
@@ -0,0 +1,43 @@
+package cash.z.ecc.sdk
+
+import android.content.Context
+import cash.z.ecc.android.bip39.Mnemonics
+import cash.z.ecc.android.bip39.toSeed
+import cash.z.ecc.android.sdk.Initializer
+import cash.z.ecc.android.sdk.Synchronizer
+import cash.z.ecc.android.sdk.tool.DerivationTool
+import cash.z.ecc.android.sdk.type.UnifiedViewingKey
+import cash.z.ecc.sdk.model.PersistableWallet
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+// Synchronizer needs a Companion object
+// https://github.com/zcash/zcash-android-wallet-sdk/issues/310
+object SynchronizerCompanion {
```

Comment:
The naming of SynchronizerCompanion is arbitrary, but intended to signal effectively static extension methods for the Synchronizer.

There's two parts to how this could evolve:
 - An empty companion in the SDK allows us to optionally write extension functions in consumers of the SDK.  This doesn't really change much for SDK consumers.  The code would literally be `companion object` in the SDK, then we eliminate the `object SynchronizerCompanion` in the app (instead being `fun Synchronizer.method()`).  This better organizes the extensions on the Synchronizer object, instead of having an external utility class. Depending on timing, a new Kotlin language feature might eliminate the need for that.  https://youtrack.jetbrains.com/issue/KT-11968?_gl=1*gcdmna*_ga*Nzg0MzY4ODQuMTYyOTgxMDUxMA..*_ga_0WQ2ZF5VGT*MTYzNjY1ODg0OS4zLjAuMTYzNjY1ODg0OS42MA..&_ga=2.91355738.1115415687.1636584065-78436884.1629810510
 - The code I wrote here for this PR could stay in the app or be migrated to the SDK, depending on whether we find it useful for other SDK consumers.

Reply:
Also to explicitly answer your questions: no SDK clients are not obligated to implement anything and there's no lifecycle requirement.

The empty companion in Kotlin allows for the Swift equivalent of adding a class extension method.